tutorials/intermediate/arranging_axes.py

```diff
@@ -0,0 +1,371 @@
+                              gridspec_kw=gs_kw, figsize=(4.5, 3.5),
+                              constrained_layout=True)
+for k in axd.keys():
+    annotate_axes(axd[k], f'axd["{k}"]', fontsize=14)
+fig.suptitle('plt.subplot_mosaic()')
+
+############################################################################
+# Nested Axes layouts
+# -------------------
+#
+# Sometimes it is helpful to have two or more grids of Axes that
+# may not need to be related to one another.  The most simple way to
+# accomplish this is to use `.Figure.subfigures`.  Note that the alignement
+# of the subfigure layouts is independent with the Axes spines in each
+# subfigure having independent positions.  See below for a more verbose
+# way to acheive the same effect with `~.gridspec.GridSpecFromSubplotSpec`.
+
+fig = plt.figure(constrained_layout=True)
+subfigs = fig.subfigures(1, 2, wspace=0.07, width_ratios=[1.5, 1.])
+axs0 = subfigs[0].subplots(2, 2)
+subfigs[0].set_facecolor('0.9')
+subfigs[0].suptitle('subfigs[0]\nLeft side')
+subfigs[0].supxlabel('xlabel for subfigs[0]')
+
+axs1 = subfigs[1].subplots(3, 1)
+subfigs[1].suptitle('subfigs[1]')
+subfigs[1].supylabel('ylabel for subfigs[1]')
+
+############################################################################
+# It is also possible to nest Axes using `~.pyplot.subplot_mosaic` using
+# nested lists.  This method does not use subfigures, like above, so lacks
+# the ability to add per-subfigure ``suptitle`` and ``supxlabel``, etc.
+# Rather it is a conveneince wrapper around the `~.SubplotSpec.subgridspec`
+# method described below.
+
+inner = [['innerA'],
+         ['innerB']]
+outer = [['upleft',  inner],
+          ['lowleft', 'lowright']]
+
+fig, axd = plt.subplot_mosaic(outer, constrained_layout=True)
+for k in axd.keys():
+    annotate_axes(axd[k], f'axd["{k}"]')
+
+############################################################################
+# Low-level and advanced grid methods
+# ===================================
+#
+# Internally, the arrangement of a grid of Axes is controlled by creating
+# instances of `~.GridSpec` and `~.SubplotSpec`. *GridSpec* defines a
+# (possibly non-uniform) grid of cells. Indexing into the *GridSpec* returns
+# a SubplotSpec that covers one or more grid cells, and can be used to
+# specify the location of an Axes.
+#
+# The following examples show how to use low-level methods to arrange Axes
+# using *GridSpec* objects.
+#
+# Basic 2x2 grid
+# --------------
+#
+# We can accopmplish a 2x2 grid in the same manner as
+# ``plt.subplots(2, 2)``:
+
+fig = plt.figure(figsize=(4.5, 3.5), constrained_layout=True)
+spec = fig.add_gridspec(ncols=2, nrows=2)
+ax0 = fig.add_subplot(spec[0, 0])
+annotate_axes(ax0, 'ax0')
+ax1 = fig.add_subplot(spec[0, 1])
+annotate_axes(ax1, 'ax1')
+ax2 = fig.add_subplot(spec[1, 0])
+annotate_axes(ax2, 'ax2')
+ax3 = fig.add_subplot(spec[1, 1])
+annotate_axes(ax3, 'ax3')
+fig.suptitle('Manually added subplots using add_gridspec')
```

Comment:
I would split this for some clarity:

```suggestion
spec = fig.add_gridspec(ncols=2, nrows=2)

ax0 = fig.add_subplot(spec[0, 0])
annotate_axes(ax0, 'ax0')

ax1 = fig.add_subplot(spec[0, 1])
annotate_axes(ax1, 'ax1')

ax2 = fig.add_subplot(spec[1, 0])
annotate_axes(ax2, 'ax2')

ax3 = fig.add_subplot(spec[1, 1])
annotate_axes(ax3, 'ax3')

fig.suptitle('Manually added subplots using add_gridspec')
```